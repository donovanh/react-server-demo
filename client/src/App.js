import React, { Component } from 'react'
import logo from './logo.svg'
import './App.css'
import { connect } from 'react-redux'
import { setMessage } from './store/actions'

class App extends Component {
  componentDidMount () {
    if (!this.props.message) {
      this.props.updateMessage("Hi, I'm from client!")
    }
  }

  render () {
    return (
      <div className='App'>
        <header className='App-header'>
          <img src={logo} className='App-logo' alt='logo' />
          <h1 className='App-title'>Welcome to React</h1>
        </header>
        <p>
            Redux: { this.props.message }
        </p>
      </div>
    )
  }
}

export default connect(
  ({ app }) => ({
    message: app.message
  }),
  {
    updateMessage: setMessage
  }
)(App)
