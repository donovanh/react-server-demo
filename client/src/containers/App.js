import React, { Component } from 'react'
import { connect } from 'react-redux'
import { setMessage } from '../store/appReducer'
import Heading from '../components/Heading'
import globalStyles from '../theme/globalStyles'
import ThemeProvider from '../theme/ThemeProvider'

class App extends Component {
  componentDidMount () {
    if (!this.props.message) {
      this.props.updateMessage("Hi, I'm from client!")
    }
  }

  render () {
    globalStyles()
    return (
      <ThemeProvider>
        <div className='App'>
          <Heading>Redux: { this.props.message }</Heading>
        </div>
      </ThemeProvider>
    )
  }
}

export default connect(
  ({ app }) => ({
    message: app.message
  }),
  dispatch => ({
    updateMessage: (txt) => dispatch(setMessage(txt))
  })
)(App)
