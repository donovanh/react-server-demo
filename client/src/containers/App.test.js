import React from 'react'
import { App } from './App'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'

it('renders without crashing', () => {
  const wrapper = shallow(<App message={'Hello'} />)
  expect(toJson(wrapper)).toMatchSnapshot()
})
