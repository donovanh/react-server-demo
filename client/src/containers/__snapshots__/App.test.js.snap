// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<ThemeProvider>
  <div
    className="App"
  >
    <Heading>
      Redux: 
      Hello
    </Heading>
  </div>
</ThemeProvider>
`;
